dag-sync.sh

```diff
@@ -20,7 +20,13 @@ git fetch origin
 new=$(git rev-list --reverse --topo-order HEAD..origin/main | head -1)
 # If there is no new commit hash to move to, nothing has changed, quit early
 [ -z "$new" ] && exit
-# Move ahead to this new commit
+
+# Verify if have /dags/ in the last commit
+have_dag=$(git log -p -1 origin/main  --pretty=format: --name-only | grep "/dags/")
+# If there is no files under /dags/ folder, no need to notify, quit early
+![ "$have_dag" ] && exit
```

Comment:
This may not work as expected if several PRs are merged in quick succession - `origin/main` will be several commits ahead, but the next commit being merged (defined on line 20) will only be one commit ahead. In cases where the next commit doesn't have a DAGs change but the tip of `origin/main` does, this would incorrectly report the next commit (and all commits unconditionally until the codebase had caught up with `origin/main`). Similarly, if the tip of `origin/main` doesn't have DAG changes, no commits along the way would be reported even if they did have DAG changes. I tested this locally, and I think what we'll want is:
```suggestion
# Verify if have /dags/ in the last commit
have_dag=$(git log -p -1 $new  --pretty=format: --name-only | grep "/dags/")
# If there is no files under /dags/ folder, no need to notify, quit early
[ -z "$have_dag" ] && exit
```

Reply:
Thanks, @AetherUnbound! It makes total sense.